File: Backend/Events/Converters/DigimonBattleConverter.cs
using Backend.Domain.Models;
using Backend.Events.Converters.Battles;
using Backend.Events.DTO;

namespace Backend.Events.Converters;

public static class DigimonBattleConverter
{
    public static DigimonBattleDTO ToDTO(DigimonBattle digimonBattle) => new()
    {
        Field = digimonBattle.Field,
        Enemy = digimonBattle.Enemy != null ? EnemyConverter.ToDTO(digimonBattle.Enemy) : null
    };
}

File: Frontend/src/events/converters/digimon-battle.converter.ts
import type { DeepRequired } from "@/events/dto/deep-required";
import type { DigimonBattleDTO } from "@/events/dto/digimon-battle.dto";
import type { DigimonBattle } from "@/models/digimon-battle";
import { EnemyConverter } from "./battles/enemy.converter";

export class DigimonBattleConverter {
  public static convert(digimonBattleDto: DeepRequired<DigimonBattleDTO>): DigimonBattle {
    return {
      field: digimonBattleDto.field,
      enemy: digimonBattleDto.enemy ? EnemyConverter.convert(digimonBattleDto.enemy) : null,
    };
  }
}
